src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,106 @@
+package balancer
+
+import (
+	"math/rand"
+)
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers, func(i, j int) bool {
+		mi := dialers[i].metrics()
+		mj := dialers[j].metrics()
+		return (mi.consecSuccesses - mi.consecFailures) >
```

Comment:
Based on the logic in dialer.go, either consecSuccesses or consecFailures will be 0.  So when doing this comparison, you're really just saying:

1. If only one of the dialers succeeded on its last attempt, use that otherwise
2. Use the dialer that's had the most consecutive successes.

Is that what you intended?  If so, it might be clearer to express the logic that way, e.g.:

```golang
return mi.consecSuccesses > mj.consecSuccesses
```